tests/v2/validation/fleet/provisioning/airgap/airgap.go

```diff
@@ -0,0 +1,180 @@
+package airgap
+
+import (
+	"crypto/x509"
+	"errors"
+	"os"
+	"strings"
+
+	"encoding/pem"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	corralBastionIP  = "bastion_ip"
+	corralSSHUser    = "aws_ssh_user"
+	corralPrivateKey = "corral_private_key"
+	systemRegistry   = "system-default-registry"
+
+	fleetExampleFolderName = "fleet-examples.git"
+	rancherGithubURL       = "https://github.com/rancher/"
+	fleetDefaultNS         = "fleet-default"
+
+	sshPort = ":22"
+	rsa     = "RSA"
+	tcp     = "tcp"
+)
+
+func setupAirgapFleetResources(bastionUser, bastionIP, privateKey string) (string, error) {
```

Comment:
another question here. if we are going to have more tests in future wrt airgap and fleet, it would be nice to have these methods in this file in actions so they are accessible easily.

Reply:
I don't think this would ever be shared with other tests. maybe the design discussion next week will help resolve this but for now, I think its safest to keep it local to the test, not as an action.

Reply:
correct.